rust/headless-client/src/ipc_service.rs

```diff
@@ -419,15 +424,20 @@ impl<'a> Handler<'a> {
                 self.ipc_tx
                     .send(&ServerMsg::ClearedLogs(result.map_err(|e| e.to_string())))
                     .await
-                    .context("Error while sending IPC message")?
+                    .context("Error while sending IPC message")?;
             }
             ClientMsg::Connect { api_url, token } => {
                 let token = secrecy::SecretString::from(token);
-                let result = self.connect_to_firezone(&api_url, token);
-                self.ipc_tx
-                    .send(&ServerMsg::ConnectResult(result))
+                let connect_result = self.connect_to_firezone(&api_url, token);
+                let ipc_result = self
+                    .ipc_tx
+                    .send(&ServerMsg::ConnectResult(connect_result.clone()))
```

Comment:
Cloneable errors can be a bit of a hassle. The caller of `handle_ipc_msg` just logs the error, so we technically only need a reference here. If we want to log the connect result in the IPC service, could we just log it before we send it and only treat the errors returned from this function as "meta" errors that concern themselves with the IPC operations in general?

Reply:
Alternatively, you could add a function like `send_msg` that has conditional code deciding on which level a certain message is logged.

For example, by default, each IPC message would be logged on debug, but if it is a `ConnectResult(Err(_))`, we log it on `warn`?

Reply:
Hm true.

I reverted everything and just logged the connection error directly inside `handle_ipc_msg`, so the diff is like 3 lines now